executor/envd/api/routes.py

```
@@ -187,10 +196,282 @@ async def upload_file(
+        """
+        files: list[WorkspaceFile] = []
+        total_files = 0
+        total_size = 0
+
+        try:
+            entries = sorted(
+                root_path.iterdir(), key=lambda x: (not x.is_dir(), x.name.lower())
+            )
+        except PermissionError:
+            return files, total_files, total_size
+        except Exception as e:
```

Ruff (0.14.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)